src/tracer.js

```diff
@@ -231,13 +231,23 @@ export default class Tracer {
             ctx.traceId = parent.traceId;
             ctx.spanId = Utils.getRandom64();
             ctx.parentId = parent.spanId;
-            ctx.flags = parent.flags;
-
             // reuse parent's baggage as we'll never change it
             ctx.baggage = parent.baggage;
+            ctx.flags = parent.flags;
 
-            parent.finalizeSampling();
-            ctx.finalizeSampling();
+            // Parent exists without a sampling decision
+            // An example of where this happens is in mobile clients where sampler parameters
+            // are not propagated
+            if (parent.flags & constants.DEFERRED_SAMPLING_MASK) {
+                ctx.flags ^=  constants.DEFERRED_SAMPLING_MASK;
+                if (this._sampler.isSampled(operationName, internalTags)) {
+                    ctx.flags |= constants.SAMPLED_MASK;
+                    internalTags[constants.DEFERRED_SAMPLING_TAG_KEY] = true;
+                }
+            } else {
+                parent.finalizeSampling();
+                ctx.finalizeSampling();
```

Comment:
I don't think they are the same thing. 

I was also looking at @black-adder's comment about this scheme not working with the rate limiting sampler (or in the more general case, any sampler that isn't consistent). The previous suggestion of modifying the parent span solves this problem. 

The `finalizeSampling` mechanism resets the sampling decision by calling `IsSampled` the first time `setOperation` is called.  Any subsequent `setOperation` calls are ignored.

Further, the `finalizeSampling` mechanism is only invoked for new root spans. (Anything extracted/injected is finalized in the extractor/injector)

I feel that @black-adder's suggestion might be the cleanest way of sampling non root spans in the middle of a trace.